### ISSUE TITLE
Brave uses Wayland icon instead of Brave's icon on flatpak

### ISSUE THREAD

**Issue by Angxddeep** [brave/brave-browser#41439]
### Description

So, I am facing this issue on flatpak version of brave, it shows the Wayland icon instead of Brave's icon as I have set Preferred Ozone platform as Wayland and Brave is running in a Wayland session, I have also installed the rpm version of Brave and on that it shows Brave's icon properly. 

## I am using:-
KDE PLASMA 6.1.5
Fedora 40 (Wayland session)

I also tried creating an issue on [Fedora's discussion](https://discussion.fedoraproject.org/t/wayland-icon-instead-of-application-icons-in-task-switcher/132146) forum and there I came to know that it may have something to do with upstream application itself, therefore I am reporting it here. This is also an issue with many Electron applications, but on some application it does show the correct icon, so it is kind of inconsistent. Main thing to note is that on the rpm version there is no issue, so it has something to do with flathub manifest itself

## Screenshots:-
![image](https://github.com/user-attachments/assets/6c5454bf-f31c-42bb-8ba5-d215749e80f6)
![image](https://github.com/user-attachments/assets/961b4556-76e9-4a38-8beb-0b331e1d117e)

### Steps to reproduce

1. Install Brave flatpak
2. Set Preferred Ozone platform as Wayland

### Actual result

Shows Wayland icon

### Expected result

It should show Brave's icon instead of Wayland

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

Brave	1.70.123 Chromium: 129.0.6668.89 (Official Build) unknown (64-bit) 
Revision	7c1bbf03546a64d969d010808e4f5c99fdf343bb
OS	Linux

### Channel information

- [X] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

### Miscellaneous information

It could be a KDE specific issue, although I am not sure if that's the case.

**Comment by rebron** [comment]
cc: @wknapik @fmarier

**Comment by rany2** [comment]
https://github.com/refi64/cobalt/pull/10 might fix it.

**Comment by Angxddeep** [comment]
> [refi64/cobalt#10](https://github.com/refi64/cobalt/pull/10) might fix it.

Yeah, I think it will, but it has been open for 3 months now, still the maintainer hasn't merger it.

**Comment by rany2** [comment]
@Angxddeep In the meantime, can you confirm that setting environ `CHROME_DESKTOP` to `com.brave.Browser.desktop` fixes it? You can do this with `flatpak run --env=CHROME_DESKTOP=com.brave.Browser.desktop com.brave.Browser`.

**Comment by Angxddeep** [comment]
> flatpak run --env=CHROME_DESKTOP=com.brave.Browser.desktop com.brave.Browser

No, this doesn't fix it.

**Comment by rany2** [comment]
Did you kill old instances of Brave before running it?

**Comment by Angxddeep** [comment]
> Did you kill old instances of Brave before running it? (`flatpak kill com.brave.Browser`)

Yeah, I did

**Comment by Angxddeep** [comment]
@rany2 This issue doesn't persist in ungoogled chromium.

![Image](https://github.com/user-attachments/assets/355041b0-c9c6-436e-9889-26a955cfa8e6)

**Comment by rany2** [comment]
I've actually fixed this issue upstream in Cobalt. I think this issue should be closed.

**Comment by Angxddeep** [comment]
Also, there is a protocol called [xdg_toplevel_icon](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/269) that has been merged in Wayland protocols repo and in window manager like [Kwin](https://invent.kde.org/plasma/kwin/-/merge_requests/5680) which is used by KDE Plasma. The update will come around Feb of 2025 with KDE Plasma 6.3 release. Although I think it will be merged in upstream chromium, so I don't think Brave will have to do anything with it.

As for this current bug,

> The commit here fixed the same issue for another program: [flathub/com.super_productivity.SuperProductivity@940a4e5](https://github.com/flathub/com.super_productivity.SuperProductivity/commit/940a4e5f8d12b777dc7a4fb855b9e36cb540c1b0)

I think this is the fix

**Comment by rany2** [comment]
That fix you sent is equivalent to what I've asked to do previously, specifically setting CHROME_DESKTOP.

I wonder if Brave renamed that to BRAVE_DESKTOP, the odd thing is that the fix I implemented in Chromium and Ungoogled Chromium did actually fix it for you.

Does `flatpak run --env=BRAVE_DESKTOP=com.brave.Browser.desktop com.brave.Browser` work?

**Comment by Angxddeep** [comment]
> Does `flatpak run --env=BRAVE_DESKTOP=com.brave.Browser.desktop com.brave.Browser` work?
The issue still persists

**Comment by Angxddeep** [comment]
So this is probably a KDE issue because this doesn't happen in gnome

**Comment by Angxddeep** [comment]
@rany2 This got fixed with the v1.77.95 (Apr 2, 2025) update, I think it got fixed because of the #45124 update, there are Two commits in chromium for Wayland in this changelog

https://chromium.googlesource.com/chromium/src/+/48346e13becfe512caf3ccc4e2258c490d1a2d15
https://chromium.googlesource.com/chromium/src/+/b1176ff8ff4eea4308bf50492ca457661efc17ec

These might be the reason for the fix here.
